aspnet/fundamentals/routing/sample/RoutingSample/HelloExtensions.cs

```diff
@@ -0,0 +1,20 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace RoutingSample
+{
+    public static class HelloExtensions
+    {
+        public static IRouteBuilder AddHelloRoute(this IRouteBuilder routeBuilder,
+            IApplicationBuilder app)
+        {
+            routeBuilder.Routes.Add(new Route(new HelloRouter(),
```

Comment:
This might needs a little more explanation about why both `Route` and `HelloRoute` is needed, and what the data flow is. I think it's an ok example, but needs more description.